impeller/typographer/backends/skia/typographer_context_skia.cc

```diff
@@ -117,112 +120,111 @@ static SkImageInfo GetImageInfo(const GlyphAtlas& atlas, Size size) {
+    const ISize& atlas_size,
     std::vector<Rect>& glyph_positions,
-    ISize atlas_size,
-    const std::shared_ptr<RectanglePacker>& rect_packer) {
-  TRACE_EVENT0("impeller", __FUNCTION__);
-  if (!rect_packer || atlas_size.IsEmpty()) {
-    return false;
-  }
-
-  // We assume that all existing glyphs will fit. After all, they fit before.
-  // The glyph_positions only contains the values for the additional glyphs
-  // from extra_pairs.
-  FML_DCHECK(glyph_positions.size() == 0);
-  glyph_positions.reserve(extra_pairs.size());
-  for (size_t i = 0; i < extra_pairs.size(); i++) {
-    const FontGlyphPair& pair = extra_pairs[i];
+    int64_t height_adjustment,
+    const std::shared_ptr<RectanglePacker>& rect_packer,
+    size_t start_index) {
+  FML_DCHECK(!atlas_size.IsEmpty());
 
+  for (size_t i = start_index; i < pairs.size(); i++) {
+    const auto& pair = pairs[i];
     const auto glyph_size =
         ISize::Ceil(pair.glyph.bounds.GetSize() * pair.scaled_font.scale);
     IPoint16 location_in_atlas;
     if (!rect_packer->AddRect(glyph_size.width + kPadding,   //
                               glyph_size.height + kPadding,  //
                               &location_in_atlas             //
                               )) {
-      return false;
+      return i;
     }
-    glyph_positions.emplace_back(Rect::MakeXYWH(location_in_atlas.x(),  //
-                                                location_in_atlas.y(),  //
-                                                glyph_size.width,       //
-                                                glyph_size.height       //
-                                                ));
+    glyph_positions.push_back(Rect::MakeXYWH(
+        location_in_atlas.x(),                      //
+        location_in_atlas.y() + height_adjustment,  //
+        glyph_size.width,                           //
+        glyph_size.height                           //
+        ));
   }
 
-  return true;
+  return pairs.size();
 }
 
-static ISize OptimumAtlasSizeForFontGlyphPairs(
-    const std::vector<FontGlyphPair>& pairs,
-    std::vector<Rect>& glyph_positions,
+static ISize ComputeNextAtlasSize(
     const std::shared_ptr<GlyphAtlasContext>& atlas_context,
-    const ISize& max_texture_size) {
-  static constexpr ISize kMinAtlasSize = ISize{4096, 1024};
-
-  TRACE_EVENT0("impeller", __FUNCTION__);
-
-  ISize current_size = kMinAtlasSize;
-  size_t total_pairs = pairs.size() + 1;
-  do {
-    auto rect_packer = std::shared_ptr<RectanglePacker>(
-        RectanglePacker::Factory(current_size.width, current_size.height));
+    const std::vector<FontGlyphPair>& extra_pairs,
+    std::vector<Rect>& glyph_positions,
+    size_t glyph_index_start,
+    int64_t max_texture_height) {
+  // Because we can't grow the skyline packer horizontally, pick a reasonable
+  // large width for all atlases.
+  static constexpr auto kAtlasWidth = 4096u;
+  static constexpr auto kMinAtlasHeight = 1024;
```

Comment:
Nit: Missing `u`. Especially since you are also using auto.

Reply:
Actually these should be signed since we pass them to ISize. Will remove auto.